lib/reporter.js

```diff
@@ -25,145 +25,190 @@ class AllureReporter extends events.EventEmitter {
 
         const { epilogue } = this.baseReporter
 
+        this.bindEventFunctions()
+
+        // Hook events
+        this.on('hook:start', this.hookStart)
+        this.on('hook:end', this.hookEnd)
+
+        // Test framework events
+        this.on('suite:start', this.suiteStart)
+        this.on('suite:end', this.suiteEnd)
+        this.on('test:start', this.testStart)
+        this.on('test:pass', this.testPass)
+        this.on('test:fail', this.testFail)
+        this.on('test:pending', this.testPending)
+
+        // Runner events (webdriver)
+        this.on('runner:command', this.runnerCommand)
+        this.on('runner:result', this.runnerResult)
+        this.on('runner:screenshot', this.runnerScreenshot)
         this.on('end', () => {
             epilogue.call(baseReporter)
         })
 
-        this.on('suite:start', (suite) => {
-            const allure = this.getAllure(suite.cid)
-            const currentSuite = allure.getCurrentSuite()
-            const prefix = currentSuite ? currentSuite.name + ' ' : ''
-            allure.startSuite(prefix + suite.title)
-        })
+        // Allure events
+        this.on('allure:feature', this.allureFeature)
+        this.on('allure:addEnvironment', this.allureAddEnvironment)
+        this.on('allure:attachment', this.allureAttachment)
+    }
 
-        this.on('suite:end', (suite) => {
-            this.getAllure(suite.cid).endSuite()
-        })
+    bindEventFunctions () {
+        this.hookStart = this.hookStart.bind(this)
```

Comment:
Let's do

```js
this.hookStart = ::this.hookStart
// ...
```

Reply:
This us still poc, so I am using older but stable syntax.

I am trying to revise master to my branch but Ci build failed, don't know why :/

Reply:
https://github.com/netmajor/wdio-allure-reporter/pull/4

https://travis-ci.org/netmajor/wdio-allure-reporter/builds/330083718?utm_source=github_status&utm_medium=notification

Reply:
When I tried to merge from master to my branch i get : There isn’t anything to compare.

Reply:
just switch to master, pull latest code, create a branch, apply changes and create a new PR .. I guess this is the easiest

Reply:
Ok. I will try it tomorrow.